source-code/ide-extension/src/commands/editMessage.ts

```diff
@@ -0,0 +1,58 @@
+import { query } from '@inlang/core/query'
+import { setState, state } from '../state.js'
+import { msg } from '../utilities/message.js'
+import { EventEmitter, window } from 'vscode'
+import type { Message } from '@inlang/core/ast'
+
+const onDidEditMessageEmitter = new EventEmitter<void>()
+export const onDidEditMessage = onDidEditMessageEmitter.event
+
+export const editMessageCommand = {
+  id: "inlang.editMessage",
+  title: "Inlang: Edit Message",
+  callback: async function ({ messageId, resource }: { messageId: string, resource: string }) {
+    const { writeResources } = state().config
+    const currentResource = state().resources.find((r) => r.languageTag.name === resource)
+    if (!currentResource) {
+      return msg("Couldn't retrieve resource", "warn", "notification")
+    }
+
+    const message = query(currentResource).get({ id: messageId })
+
+    const newValue = await window.showInputBox({
+      title: "Enter new value:",
+      placeHolder: message?.pattern.elements[0]!.value as string
```

Comment:
@openscript Can we also set the exisiting value here and not only the placeholder? But only if a value exists ofc. This could help resolve typos in long strings and not have to type everything again.

Reply:
Sounds good!